service/bitbucket/annotator.go

```diff
@@ -99,6 +99,7 @@ func (r *ReportAnnotator) Post(_ context.Context, c *reviewdog.Comment) error {
 // Flush posts comments which has not been posted yet.
 func (r *ReportAnnotator) Flush(ctx context.Context) error {
 	r.muAnnotations.Lock()
+	defer func() { r.comments = nil }()
 	defer r.muAnnotations.Unlock()
```

Comment:
The defer statement is executed in LIFO order.
In this case, the order is `r.muAnnotations.Unlock()`, `r.comments = nil`.
Is it intentional that `r.comments = nil` is outside the lock scope?

```suggestion
	r.muAnnotations.Lock()
	defer r.muAnnotations.Unlock()
	defer func() { r.comments = nil }()
```

Reply:
Oh, may bad. Thank you for the review!